apps/wallet/src/ui/app/components/accounts/UnlockAccountButton.tsx

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState } from 'react';
+import { UnlockAccountModal } from './UnlockAccountModal';
+import { useUnlockMutation } from '../../hooks/useUnlockMutation';
+import { Button } from '../../shared/ButtonUI';
+import { SocialButton } from '../../shared/SocialButton';
+import { type SerializedUIAccount } from '_src/background/accounts/Account';
+import { isZkAccountSerializedUI } from '_src/background/accounts/zk/ZkAccount';
+
+export type UnlockAccountButtonProps = {
+	account: SerializedUIAccount;
+	title?: string;
+};
+export function UnlockAccountButton({
+	account,
+	title = 'Unlock Account',
+}: UnlockAccountButtonProps) {
+	const { id, isPasswordUnlockable } = account;
+	const unlockMutation = useUnlockMutation();
+	const [isPasswordModalVisible, setIsPasswordModalVisible] = useState(false);
```

Comment:
this looks good for now - I have a WIP branch that uses context so we don't have to tie the modal to the unlock button.